apps/site-landing/src/containers/BenefitsSimulator/Form/Advanced.tsx

```diff
@@ -0,0 +1,276 @@
+type UseNumberState = ReturnType<typeof UseStateWrapper>;
+
+const debounceInterval = 500;
+
+export const Advanced = ({ params, setParams, years, setYears }: Props) => {
+
+  const onChangedInObj = (name: keyof SimulationParameters) =>
+    debounce((val: any) => setParams((prev: SimulationParameters) => ({
+      ...prev,
+      [name]: val,
+    })), debounceInterval);
+
+  const activeIdxState = useState(-1);
+  const dbOnSetYears = debounce(v => setYears(Math.max(1, v)), debounceInterval);
+
+  return (
+    <div className={styles.formPane} >
+      <RangeFieldAndScale
+        label={basic.startingValue}
+        scaleType="currency"
+        currency="CAD"
+        maximum={1000}
+        step={1}
+        initial={params.initialBalance}
+        onChange={onChangedInObj("initialBalance")}
+      />
+      <RangeFieldAndScale
+        label={basic.rangeDuration}
+        scaleType="unit"
+        unit="year"
+        maximum={60}
+        initial={years}
+        onChange={dbOnSetYears}
+      />
+      <Accordion>
+        <PeriodicalGroup
+          title={translations.income}
+          index={0}
+          activeIndexState={activeIdxState}
+          params={params.income}
+          setParams={onChangedInObj("income")}
+        />
+        <PeriodicalGroup
+          title={translations.credit}
+          index={1}
+          activeIndexState={activeIdxState}
+          params={params.credit}
+          setParams={onChangedInObj("credit")}
+        >
+          <RangeFieldAndScale
+            label={translations.cashBackRate}
+            scaleType="percent"
+            maximum={0.05}
+            step={0.01}
+            initial={params.credit.cashBackRate}
+            onChange={v => onChangedInObj("credit")({ ...params.credit, cashBackRate: v })}
+          />
+          <RangeFieldAndScale
+            label={translations.interestRate}
+            scaleType="percent"
+            maximum={1}
+            step={0.01}
+            initial={params.credit.interestRate}
+            onChange={v => onChangedInObj("credit")({ ...params.credit, interestRate: v })}
+          />
+
+        </PeriodicalGroup>
+        <PeriodicalGroup
+          title={translations.cash}
+          index={2}
+          activeIndexState={activeIdxState}
+          params={params.cash}
+          setParams={onChangedInObj("cash")}
+        />
+        <ShockAbsorberGroup
+          title={translations.cash}
+          index={3}
+          activeIndexState={activeIdxState}
+          params={params.shockAbsorber}
+          setParams={onChangedInObj("shockAbsorber")}
+        />
+      </Accordion>
+    </div>
+  )
+}
+
+type GroupProps<T> = {
+  title: MessageDescriptor;
+  index: number;
+  activeIndexState: UseNumberState;
+  params: T;
+  setParams: (v: T) => void;
+}
+
+const PeriodicalGroup: React.FC<GroupProps<PeriodicalParams>> = ({ title, index, params, setParams, activeIndexState, children }) => {
```

Comment:
This file actually contains 3 components.  Perhaps it should be refactored into a sub-directory and 1 file each component?

Reply:
Would be nice and I am absolutely not saying that because I spend several days fighting with a google api and that I am angry at every single developer on earth.